File: python/sum_pairs.py
def sum_pairs(ls, target):
    sums = []
    for i in range(0,len(ls)):
        for j in range(1, len(ls)):
            if(i<j):
                if(ls[i]+ ls[j] == target):
                    sums.append([ls[i], ls[j]])

    if(len(sums) == 0):
        return("unable to find pairs")
    else:
        return sums

# Big O == O(n) because hashmap only has to loop through the list once.
def hashmap_sum_pairs(ls, target):
    hashmap = {}
    sums = []

    # hash logic
    for i, n in enumerate(ls):
        diff = (target - n)
        if diff in hashmap:
            sums.append([diff, n])
            hashmap[n] = i
        else:
            hashmap[n] = i
            
    if len(sums) != 0:
        return sorted(sums)
    else:
        return("unable to find pairs")

File: js/sumPairs.js
exports.sumPairs = function(arr, target) {
    let pairs = [];

    for(let i=0; i<arr.length; i++) {
        for(let j=1; j<arr.length; j++) {
            if(i < j) {
                if((arr[i] + arr[j] === target)) {
                    pairs.push([arr[i],arr[j]])
                }
            }
        }
    }
    
    if(pairs.length != 0){
        return pairs
    } else {
        return ('unable to find pairs')
    }
};

// console.log(exports.sumPairs([1,2,3,4,5], 9));  // [[4,5]]
// console.log(exports.sumPairs([1,2,3,4,5], 7));  // [[2,5], [3,4]]
// console.log(exports.sumPairs([3, 1, 5, 8, 2], 27)); // 'unable to find pairs'

let test = exports.sumPairs([1,2,3,4,5], 9);
console.log('Function answer:', test, '->', typeof test);
console.log('Expected answer:', [[4,5]], '->', typeof ([[4,5]]));

if(test === [[4,5]]) {
    console.log('True');
} else {
    console.log('False');
}